src/duckling/entitysystem/services/entity-box.service.ts

```diff
@@ -29,11 +33,17 @@ export class EntityBoxService extends BaseAttributeService<AttributeBoundingBox>
     getAttributeBox(key : AttributeKey, entity : Entity) : any {
         let getBox = this.getImplementation(key);
         if (getBox) {
+            let box : Box2;
             if (this._asset.areAssetsLoaded(entity, key)) {
-                return getBox(entity, this._asset);
+                box = getBox(entity, this._asset);
             } else {
-                return drawnConstructBounds(drawMissingAsset(this._asset));
+                box = drawnConstructBounds(drawMissingAsset(this._asset));
             }
+
+            if (box) {
+                box.position = this._entityPosition.getPosition(this._entitySystem.getKey(entity));
```

Comment:
Don't Mutate the box! You should remove the copy in boxUnion and turn this into an immutable assign.